src/clp_ffi_py/ir/native/decoding_methods.cpp

```diff
@@ -23,30 +23,36 @@
 namespace clp_ffi_py::ir::native {
 namespace {
 /**
- * Decodes the next log event from the CLP IR buffer `decoder_buffer`. If
- * `py_query` is non-null decode until finding a log event that matches the
- * query.
+ * Decodes the next log event from the CLP IR buffer `decoder_buffer` until
+ * terminate handler returns true.
+ * @tparam TerminateHandler Method to determine if the decoding should
+ * terminate, and set the return value for termination.
+ * Signature: (
+ *         [maybe_unused] ffi::epoch_timestamp_ms timestamp,
+ *         [maybe_unused] std::string_view decoded_log_message,
+ *         [maybe_unused] size_t decoded_log_event_idx,
+ *         PyObject*& return_value
+ * ) -> bool;
```

Comment:
Why is TerminateHandler a template type and not a normal function pointer param?

Reply:
Here's a conclusion of our off-line discussion:
1. We should prefer lambda rather than C style function pointer or `std::function` since it is more lightweight.
2. We should use `concept` (C++20) to specify the function signature.